### PR TITLE
Add option to always expand posts marked with content warnings

### LINKED ISSUE #1997: Option to expand content warnings by default
This is different from "expand sensitive media by default."

### PR DESCRIPTION
**Pull request by tepreece**
Added an option to always expand posts marked with content warnings. If the option is enabled, posts with content warnings are expanded by default, but can be closed using the "show less" button as usual.

This adds a new intl key. I wasn't sure whether the correct approach was just to add the key to en_US or to add it (untranslated) to every language; I've just done the former for now, but if it does need to be added untranslated to every language please let me know and I'll do so.

Fixes #1997.

### CODE REVIEW

**Comment by nolanlawson** [comment]
LGTM, thanks!

**Comment by nolanlawson** [comment]
And no, you don't need to add it elsewhere. The other ones will just default to English.

**Review by nolanlawson — APPROVED** [review]

**Comment by tepreece** [comment]
> And no, you don't need to add it elsewhere. The other ones will just default to English.

Great, thanks!

**Comment by nolanlawson** [comment]
Sorry, but CI appears to be broken right now. #2349 . I would prefer to get tests passing before merging any PR

**Comment by tepreece** [comment]
> Sorry, but CI appears to be broken right now. #2349 . I would prefer to get tests passing before merging any PR

Makes sense - take as long as you need. It looks at first glance to be all unrelated to my PR, but if I have broken anything then please let me know and I'll fix it!

**Comment by nolanlawson** [comment]
It looks like there is a small bug with this:

1. Enable the setting
2. Go to a CW'ed post (which is now expanded by default)
3. It's expanded (good!)
4. Click "Show less"
5. It's still expanded :confused: 
6. Click "Show less" again
7. Now it's collapsed.

It should collapse on the first try. I'll look at it.

File: src/intl/en-US.js [hunk 1]
```diff
@@ -368,6 +368,7 @@ export default {
   general: 'General',
   generalSettings: 'General settings',
   showSensitive: 'Show sensitive media by default',
+  showAllSpoilers: 'Always expand posts marked with content warnings',
```
Comment (nolanlawson) [inline src/intl/en-US.js:371]:
```suggestion
  showAllSpoilers: 'Expand content warnings by default',
```

To me, this fits better alongside "Show sensitive media by default".

**Comment by nolanlawson** [comment]
Fixed it and added a test to confirm the fix: 04388ff3

File: src/routes/_components/status/Status.html [hunk 2]
```diff
@@ -260,7 +260,7 @@
         notification && notification.status &&
           notification.type !== 'mention' && notification.status.id === originalStatusId
       ),
-      spoilerShown: ({ $spoilersShown, uuid }) => !!$spoilersShown[uuid],
+      spoilerShown: ({ $spoilersShown, uuid, $showAllSpoilers }) => ($spoilersShown[uuid] === undefined ? !!$showAllSpoilers : !!$spoilersShown[uuid]),
```
Comment (nolanlawson) [inline src/routes/_components/status/Status.html:263]:
```suggestion
      spoilerShown: ({ $spoilersShown, uuid, $showAllSpoilers }) => (typeof $spoilersShown[uuid] === 'undefined' ? !!$showAllSpoilers : !!$spoilersShown[uuid]),
```

Just to match the code style I use elsewhere in the codebase

File: tests/spec/043-content-warnings.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  generalSettingsButton,
+  homeNavButton,
+  getNthStatus,
+  getNthShowOrHideButton
+} from '../utils'
+import { loginAsFoobar } from '../roles'
+import { homeTimeline } from '../fixtures.js'
+import { Selector as $ } from 'testcafe'
+
+fixture`043-spoilers.js`
```
Comment (nolanlawson) [inline tests/spec/043-content-warnings.js:15]:
```suggestion
fixture`043-content-warnings.js`
```

**Pull request merged**